src/common/variables.c

```diff
@@ -454,7 +454,7 @@ static char *variable_get_value(dt_variables_params_t *params, char **variable)
       */
       {
         char *replacement = expand(params, variable, ')');
-        if(!base_value || !*base_value)
```

Comment:
This would lead to a null pointer dereference if `base_value` would be null.

Reply:
No as in C operator || does not evaluate the second argument if first one is true. Here NULL will make the first argument true and so if won't evaluate second one.

Reply:
You are right here. Nevertheless, the original code was confusing in my view. The null pointer test can be omitted.`!base_value ` _always_ evaluates to false, see comment in line 442. I suggest to omit the test here as done in line 473.

Reply:
Agreed, the test should just be:

`if(!*base_value)`